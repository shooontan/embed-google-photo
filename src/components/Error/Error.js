// @flow
import React from 'react';

type ErrorProps = {
  message: string,
};

type CheckErrorProps = ErrorProps & {
  error: boolean,
};

const Error = ({ message }: ErrorProps) => <p>{message}</p>;

const isErrorCheck = ({ error, message }: CheckErrorProps) =>
  (error ? <Error message={message} /> : null);

export default isErrorCheck;
