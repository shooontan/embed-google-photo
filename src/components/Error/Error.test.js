// @flow
import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import Error from './Error';

test('Error Component', () => {
  let testRenderer = TestRenderer.create(<Error error={false} message="ðŸ˜ˆ" />);
  let testTree = testRenderer.toJSON();
  expect(testTree).toMatchSnapshot();

  testRenderer = TestRenderer.create(<Error error message="ðŸ’€" />);
  testTree = testRenderer.toJSON();
  expect(testTree).toMatchSnapshot();
});
