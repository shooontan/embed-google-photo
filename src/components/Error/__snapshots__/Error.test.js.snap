// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Component 1`] = `null`;

exports[`Error Component 2`] = `
<p>
  ðŸ’€
</p>
`;
